### PR TITLE
🔒️Maintenance: Fix Regex DOSes

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
After several large changes of files containing regexes, Sonarcloud found some "security hotspots".

This PR aims at reducing the risks of Regex DOS via catastrophic backtracking.
see the following reference: [https://medium.com/superhuman-co/how-to-eliminate-regular-expression-denial-of-service-f8983bb76afc](https://medium.com/superhuman-co/how-to-eliminate-regular-expression-denial-of-service-f8983bb76afc)
This proposes to use the following tool to find out whether a regex is problematic or not:
- https://regex.rip/?utm_source=medium&utm_medium=blog&utm_campaign=redos

## Related issue/s
- contains https://github.com/ITISFoundation/osparc-simcore/pull/4116
- relates to https://github.com/ITISFoundation/osparc-issues/issues/675

### CODE REVIEW

**Review by mrnicegyu11 — APPROVED** [review]
thanks a lot, looks good :)

**Review by GitHK — APPROVED** [review]
very nice 👍

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: packages/models-library/src/models_library/basic_regex.py [hunk 1]
```diff
@@ -68,8 +68,10 @@
     # NOTE: https://docs.docker.com/config/labels-custom-metadata/#key-format-recommendations
     r"^(?!(\.|\-|com.docker\.|io.docker\.|org.dockerproject\.|\d))(?!.*(--|\.\.))[a-z0-9\.-]+(?<![\d\.\-])$"
 )
-DOCKER_IMAGE_KEY_RE = r"[\w/-]+"
-DOCKER_IMAGE_VERSION_RE = r"[\w/.]+"
 DOCKER_GENERIC_TAG_KEY_RE: Final[re.Pattern] = re.compile(
-    r"^(?P<registry_host>(?:(?:(?:[a-zA-Z0-9-]+\.)+[a-zA-Z0-9-]+(?::\d+)?)|[a-zA-Z0-9-]+:\d+))?(?:/)?(?P<docker_image>(?:[a-z0-9][a-z0-9_.-]*/)*[a-z0-9-_]+[a-z0-9])(?::(?P<docker_tag>[\w][\w.-]{0,126}[\w]))?(?P<docker_digest>\@sha256:[a-fA-F0-9]{64})?$"
+    # NOTE: https://docs.docker.com/engine/reference/commandline/tag/#description
+    r"^(?:(?P<registry_host>[a-z0-9-]+(?:\.[a-z0-9-]+)+(?::\d+)?|[a-z0-9-]+:\d+)/)?"
```
Comment (pcrespov) [inline packages/models-library/src/models_library/basic_regex.py:73]: multiple line is far more readable!

**Review by pcrespov — APPROVED** [review]
:tada: Thx!

**Pull request merged**